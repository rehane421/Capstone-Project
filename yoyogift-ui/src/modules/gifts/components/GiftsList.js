// import React from "react";
// import PropTypes from "prop-types";
// import { withStyles } from "@material-ui/core/styles";
// import Table from "@material-ui/core/Table";
// import TableBody from "@material-ui/core/TableBody";
// import TableCell from "@material-ui/core/TableCell";
// import TableFooter from "@material-ui/core/TableFooter";
// import TablePagination from "@material-ui/core/TablePagination";
// import TableRow from "@material-ui/core/TableRow";
// import Paper from "@material-ui/core/Paper";
// import Grid from "@material-ui/core/Grid";
// import { Button } from "@material-ui/core";
// import { List } from "react-virtualized";
// import GiftCard from "../../common/components/GiftCard";
// import { TablePaginationActionsWrapped } from "../../common/components/TablePaginationActionsWrapped";

// const styles = theme => ({
//   root: {
//     width: "90%",
//     margin: "2%",
//     padding: "1%"
//   },
//   table: {
//     minWidth: 100
//   },
//   tableWrapper: {
//     overflow: "hidden"
//   }
// });

// export class GiftsList extends React.Component {
//   state = {
//     giftCardsFiltered: this.props.giftCardsFiltered,
//     page: 0,
//     rowsPerPage: 10
//   };

//   rowRenderer = ({
//     key, // Unique key within array of rows
//     index, // Index of row within collection
//     isScrolling, // The List is currently being scrolled
//     isVisible, // This row is visible within the List (eg it is not an overscanned row)
//     style // Style object to be applied to row (to position it)
//   }) => {
//     let { giftCardsFiltered, userDetails } = this.props;
//     return (
//       <div key={key} style={style}>
//         <Grid container spacing={16}>
//           <p>{index}</p>
//           <Grid item xs={12} sm={6} md={3} lg={3}>
//             <GiftCard
//               giftCard={giftCardsFiltered[index]}
//               userEmail={userDetails.email}
//             />
//             {userDetails.email === "lathak95@gmail.com" ||
//             this.props.userDetails.email === "yoyogiftg2@gmail.com" ? (
//               <Button
//                 variant="contained"
//                 color="primary"
//                 onClick={() =>
//                   this.props.handleUpdateClick(giftCardsFiltered[index].id)
//                 }
//               >
//                 UPDATE
//               </Button>
//             ) : null}
//           </Grid>
//         </Grid>
//       </div>
//     );
//   };

//   // handleChangePage = (event, page) => {
//   //   this.setState({ page });
//   // };

//   // handleChangeRowsPerPage = event => {
//   //   this.setState({ page: 0, rowsPerPage: event.target.value });
//   // };

//   render() {
//     // let { classes, giftCardsFiltered, userDetails } = this.props;
//     // const { rowsPerPage, page } = this.state;
//     // const emptyRows =
//     //   rowsPerPage -
//     //   Math.min(rowsPerPage, giftCardsFiltered.length - page * rowsPerPage);
//     let { giftCardsFiltered } = this.props;
//     return (
//       // <Paper className={classes.root}>
//       //   <div className={classes.tableWrapper}>
//       //     <Table className={classes.table}>
//       //       <TableBody>
//       //         <Grid container spacing={16}>
//       //           {giftCardsFiltered
//       //             .slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage)
//       //             .map(giftCard => {
//       //               return (
//       //                 <Grid item xs={12} sm={6} md={3} lg={3}>
//       //                   <GiftCard
//       //                     giftCard={giftCard}
//       //                     userEmail={userDetails.email}
//       //                   />
//       //                   {/* {(userDetails.email === "lathak95@gmail.com" || this.props.userDetails.email === "yoyogiftg2@gmail.com")? <Button variant="contained" color="primary" onClick={()=> this.props.handleUpdateClick(giftCard.id)}>UPDATE</Button> : null} */}
//       //                 </Grid>
//       //               );
//       //             })}
//       //         </Grid>
//       //         {emptyRows > 0 && (
//       //           <TableRow style={{ height: 48 * emptyRows }}>
//       //             <TableCell colSpan={6} />
//       //           </TableRow>
//       //         )}
//       //       </TableBody>
//       //       <TableFooter>
//       //         <TableRow>
//       //           <TablePagination
//       //             rowsPerPageOptions={[10, 20]}
//       //             colSpan={3}
//       //             count={giftCardsFiltered.length}
//       //             rowsPerPage={rowsPerPage}
//       //             page={page}
//       //             SelectProps={{
//       //               native: true
//       //             }}
//       //             onChangePage={this.handleChangePage}
//       //             onChangeRowsPerPage={this.handleChangeRowsPerPage}
//       //             ActionsComponent={TablePaginationActionsWrapped}
//       //             props
//       //           />
//       //         </TableRow>
//       //       </TableFooter>
//       //     </Table>
//       //   </div>
//       // </Paper>
//       <div alignItems="center">
//         <List
//           width={1000}
//           height={600}
//           rowHeight={315}
//           rowRenderer={this.rowRenderer}
//           rowCount={giftCardsFiltered.length}
//         />
//       </div>
//     );
//   }
// }

// GiftsList.propTypes = {
//   classes: PropTypes.object.isRequired
// };

// export default withStyles(styles)(GiftsList);

import React from "react";
import PropTypes from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import Table from "@material-ui/core/Table";
import TableBody from "@material-ui/core/TableBody";
import TableCell from "@material-ui/core/TableCell";
import TableFooter from "@material-ui/core/TableFooter";
import TablePagination from "@material-ui/core/TablePagination";
import TableRow from "@material-ui/core/TableRow";
import Paper from "@material-ui/core/Paper";
import Grid from "@material-ui/core/Grid";
import { List } from "react-virtualized";
import { Button } from "@material-ui/core";
import GiftCard from "../../common/components/GiftCard";
import { TablePaginationActionsWrapped } from "../../common/components/TablePaginationActionsWrapped";

const styles = theme => ({
  root: {
    width: "90%",
    margin: "2%",
    padding: "1%"
  },
  table: {
    minWidth: 100
  },
  tableWrapper: {
    overflow: "hidden"
  }
});

class GiftsList extends React.Component {
  state = {
    giftCardsFiltered: this.props.giftCardsFiltered,
    searchedCard: this.props.searchedCard,
    page: 0,
    rowsPerPage: 10
  };
  rowRenderer = ({
    key, // Unique key within array of rows
    index, // Index of row within collection
    isScrolling, // The List is currently being scrolled
    isVisible, // This row is visible within the List (eg it is not an overscanned row)
    style // Style object to be applied to row (to position it)
  }) => {
    let { giftCardsFiltered, searchedCard, userDetails } = this.props;
    return (
      <div key={key} style={style}>
        {searchedCard
          // .slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage)
          .map(giftCard => {
            return (
              <Grid item xs={12} sm={6} md={3} lg={3}>
                <GiftCard giftCard={giftCard} userEmail={userDetails.email} />
                {/* {(userDetails.email === "lathak95@gmail.com" || this.props.userDetails.email === "yoyogiftg2@gmail.com")? <Button variant="contained" color="primary" onClick={()=> this.props.handleUpdateClick(giftCard.id)}>UPDATE</Button> : null} */}
              </Grid>
            );
          })}
      </div>
    );
  };

  handleChangePage = (event, page) => {
    this.setState({ page });
  };

  handleChangeRowsPerPage = event => {
    this.setState({ page: 0, rowsPerPage: event.target.value });
  };

  render() {
    let { classes, giftCardsFiltered, userDetails, searchedCard } = this.props;
    // const { rowsPerPage, page } = this.state;
    // const emptyRows =
    //   rowsPerPage -
    //   Math.min(rowsPerPage, searchedCard.length - page * rowsPerPage);
    return (
      <Paper className={classes.root}>
        <div className={classes.tableWrapper}>
          <Table className={classes.table}>
            <TableBody>
              <Grid container spacing={16}>
                <List
                  width={1000}
                  height={600}
                  rowHeight={315}
                  rowRenderer={this.rowRenderer}
                  rowCount={giftCardsFiltered.length}
                />
              </Grid>
              {/* {emptyRows > 0 && (
                <TableRow style={{ height: 48 * emptyRows }}>
                  <TableCell colSpan={6} />
                </TableRow>
              )} */}
            </TableBody>
            {/* <TableFooter>
              <TableRow>
                <TablePagination
                  rowsPerPageOptions={[10, 20]}
                  colSpan={3}
                  count={giftCardsFiltered.length}
                  rowsPerPage={rowsPerPage}
                  page={page}
                  SelectProps={{
                    native: true
                  }}
                  onChangePage={this.handleChangePage}
                  onChangeRowsPerPage={this.handleChangeRowsPerPage}
                  ActionsComponent={TablePaginationActionsWrapped}
                  props
                />
              </TableRow>
            </TableFooter> */}
          </Table>
        </div>
      </Paper>
    );
  }
}

GiftsList.propTypes = {
  classes: PropTypes.object.isRequired
};

export default withStyles(styles)(GiftsList);
