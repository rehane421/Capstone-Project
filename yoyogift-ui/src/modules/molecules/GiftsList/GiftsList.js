import React from "react";
import PropTypes from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import GiftCard from "../../atoms/GiftCard/GiftCard";
import { List, AutoSizer } from "react-virtualized";
const styles = theme => ({
  root: {
    width: "20%",
    margin: "1%",
    padding: "1%",
    float: "left",
    [theme.breakpoints.down("sm")]: {
      // padding: "7%",
      width: "30%"
    },
    [theme.breakpoints.down("xs")]: {
      padding: "7%",
      width: "80%"
    }
  },
  table: {
    minWidth: 100
  },
  tableWrapper: {
    overflow: "hidden"
  }
});

export class GiftsList extends React.PureComponent {
  state = {
    searchedCard: this.props.searchedCard
  };
  render() {
    let { searchedCard, userDetails, classes } = this.props;

    return (
      <div style={{ marginTop: "10px", height: "80vh" }}>
        <AutoSizer>
          {({ height, width }) => {
            const itemsPerRow = Math.floor(width / 300) || 1;
            const rowCount = Math.ceil(searchedCard.length / itemsPerRow);
            return (
              <div>
                <List
                  width={width}
                  height={height}
                  rowCount={rowCount}
                  rowHeight={340}
                  rowRenderer={({ index, key, style }) => {
                    const items = [];
                    const fromIndex = index * itemsPerRow;
                    const toIndex = Math.min(
                      fromIndex + itemsPerRow,
                      searchedCard.length
                    );

                    for (let i = fromIndex; i < toIndex; i++) {
                      items.push(
                        <div className={classes.root} key={i}>
                          <GiftCard
                            giftCard={searchedCard[i]}
                            userEmail={userDetails.email}
                          />
                        </div>
                      );
                    }

                    return (
                      <div className={classes.item} key={key} style={style}>
                        {items}
                      </div>
                    );
                  }}
                />
              </div>
            );
          }}
        </AutoSizer>
      </div>
    );
  }
}

GiftsList.propTypes = {
  classes: PropTypes.object.isRequired
};

export default withStyles(styles)(GiftsList);
