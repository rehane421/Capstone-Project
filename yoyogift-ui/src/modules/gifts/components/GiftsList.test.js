import React from "react";
import { shallow } from "../../../enzyme";
import { shallowToJson } from "enzyme-to-json";
import { GiftsList } from "./GiftsList";

describe("GiftsList", () => {
  let wrapper, props;

  props = {
    giftCardsFiltered: [{ id: 1 }],
    classes: {
      root: "root"
    },
    userDetails: {
      email: "yoyogiftg2@gmail.com"
    },
    handleUpdateClick: jest.fn()
  };
  let props2 = {
    giftCardsFiltered: [{ id: 1 }],
    classes: {
      root: "root"
    },
    userDetails: {
      email: "udha2@gmail.com"
    },
    handleUpdateClick: jest.fn()
  };
  let methodParam = { key: {}, index: 0, isScrolling: true };
  it("should render GiftsList", () => {
    wrapper = shallow(<GiftsList {...props} />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
    shallowToJson(wrapper)
      .children[0].props.rowRenderer(methodParam)
      .props.children.props.children[1].props.children[1].props.onClick();
    shallowToJson(wrapper).children[0].props.rowRenderer(methodParam);
  });

  //else part
  it("should render GiftsList", () => {
    wrapper = shallow(<GiftsList {...props2} />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
