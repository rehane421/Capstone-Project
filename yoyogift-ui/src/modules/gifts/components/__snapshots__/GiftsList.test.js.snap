// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GiftsList should render GiftsList 1`] = `
<div
  alignItems="center"
>
  <List
    autoHeight={false}
    estimatedRowSize={30}
    height={600}
    noRowsRenderer={[Function]}
    onRowsRendered={[Function]}
    onScroll={[Function]}
    overscanIndicesGetter={[Function]}
    overscanRowCount={10}
    rowCount={1}
    rowHeight={315}
    rowRenderer={[Function]}
    scrollToAlignment="auto"
    scrollToIndex={-1}
    style={Object {}}
    width={1000}
  />
</div>
`;

exports[`GiftsList should render GiftsList 2`] = `
<div
  alignItems="center"
>
  <List
    autoHeight={false}
    estimatedRowSize={30}
    height={600}
    noRowsRenderer={[Function]}
    onRowsRendered={[Function]}
    onScroll={[Function]}
    overscanIndicesGetter={[Function]}
    overscanRowCount={10}
    rowCount={1}
    rowHeight={315}
    rowRenderer={[Function]}
    scrollToAlignment="auto"
    scrollToIndex={-1}
    style={Object {}}
    width={1000}
  />
</div>
`;
